src/components/utils/rhf-inputs/chip-items-input.js

```diff
@@ -0,0 +1,134 @@
+        name,
+    });
+
+    const addItem = useCallback(
+        (value) => {
+            // check if element is already present
+            if (getValues(name).find((v) => v === value) !== undefined) {
+                snackError({
+                    messageTxt: '',
+                    headerId: 'ElementAlreadyUsed',
+                });
+            } else {
+                append(value);
+            }
+        },
+        [append, getValues, snackError, name]
+    );
+
+    const keyPress = (e) => {
+        if (e.keyCode === 13 && textEntered.length > 0) {
+            addItem(textEntered);
+            setTextEntered('');
+        }
+    };
+
+    const handleChange = (e) => {
+        setTextEntered(e.target.value);
+    };
+
+    return (
+        <>
+            <FormControl
+                className={clsx(classes.formChipItems1, {
+                    [classes.formChipItemsError]: error?.message,
+                })}
+                error={!!error?.message}
+            >
+                {elements?.length === 0 && label && (
+                    <FieldLabel
+                        label={label}
+                        optional={
+                            !isFieldRequired(
+                                name,
+                                validationSchema,
+                                getValues()
+                            )
+                        }
+                    />
+                )}
+                {elements?.length > 0 && (
+                    <FormControl className={classes.formChipItems2}>
+                        {elements.map((item, index) => (
+                            <Chip
+                                key={item.id}
+                                size="small"
+                                onDelete={() => {
+                                    remove(index);
+                                }}
+                                label={
+                                    <OverflowableText
+                                        text={
+                                            <RawReadOnlyInput
+                                                name={`${name}.${index}`}
+                                            />
+                                        }
+                                        style={{ width: '100%' }}
+                                    />
+                                }
+                            />
+                        ))}
+                    </FormControl>
+                )}
+
+                <TextField
+                    variant="standard"
+                    InputProps={{
+                        disableUnderline: true,
+                        style: {
+                            height: '30px',
+                            left: '5px',
+                        },
+                    }}
+                    value={textEntered}
+                    onKeyDown={keyPress}
+                    onChange={handleChange}
```

Comment:
maybe onBlur here to add an item without Enter ?

Reply:
Done